File: adminportal/src/Dto/AdminPortal/AddUserRequest.ts
export type AddUserRequest = {
    //Most of these props can be generated in the backend code. What we do need from the admin frontend is the username, email address, password and their roleId, else default that to "User" if not provided by the frontend.
    username: string,
    password: string,
    emailAddress: string,
    roleId?: string,
}

File: server/RestApiServer.Endpoints/Dto/Admin/AddUserRequest.cs
namespace RestApiServer.Endpoints.Dto.Admin
{
    public class AddUserRequest
    {
        public required string Username { get; set; }
        public required string Password { get; set; }
        public required string EmailAddress { get; set; }
        public required string RoleId { get; set; }
    }
}